src/components/postgresConnect.ts

```diff
@@ -360,6 +360,46 @@ class PostgresDB {
     }
   }
 
+  public checkIfInFileTable = async (subPath: string) => {
+    const text = `
+    SELECT id 
+    FROM file
+    WHERE file_path = $1
+    `
+    try {
+      const result = await this.query({ text, values: [subPath] })
+      return result.rows.length === 0
```

Comment:
Should really be `return result.rows.length !== 0`. Otherwise it's backwards -- returns `false` when the file IS there and vice versa. So either return the opposite, or the name should be `checkIfNotInFileTable`. (I went with the former)